solr/core/src/java/org/apache/solr/core/SolrResourceLoader.java

```diff
@@ -181,6 +181,7 @@ public SolrResourceLoader(Path instanceDir, ClassLoader parent, Properties prope
     this.classLoader = URLClassLoader.newInstance(new URL[0], parent);
   }
 
+  private static HashMap<List<URL>, URLClassLoader> classLoaderCache = new HashMap<>();
```

Comment:
Is it a good idea to keep a `List<>` as a key?. Probably we should het a hash of all the URLs and use that as a key

Reply:
I think that will have similar effect as following List's hashcode ? Also, its onetime for core.
` 
public int hashCode() {
        int hashCode = 1;
        for (E e : this)
            hashCode = 31*hashCode + (e==null ? 0 : e.hashCode());
        return hashCode;
    }
`

Reply:
`List` is an `interface`. Different impls may implement `hashCode()` differently

Reply:
Here is the default [ implementation from AbstractList](http://hg.openjdk.java.net/jdk8/jdk8/jdk/file/jdk8-b132/src/share/classes/java/util/AbstractList.java#l538). Agree, specific  types(ex, int, boolean, etc) can have its own implementation. But, in general it collects hashcode of all the list elements. Similarly, it does for equal methods. 

Are you suggesting create WrapperURLList class and cache hashcode of all the urls? URL class already cache their hashcode, so I don't see much overhead there. Also, as mentioned its one time activity.

Reply:
>Are you suggesting create WrapperURLList 

That would be ideal

Reply:
I would recommend one more thing. 

Sort the URLs before computing the hash codes. 

`<path1>`, `<path2>` is equivalent to `<path2>`, `<path1>`